File: src/main/java/io/stouder/adonis/notifier/AdonisRcUpdateNotifier.java
package io.stouder.adonis.notifier;

import com.intellij.util.messages.Topic;
import io.stouder.adonis.cli.json.ace.Command;

import java.util.Map;
import java.util.Optional;

public interface AdonisRcUpdateNotifier {
    Topic<AdonisRcUpdateNotifier> ADONIS_RC_UPDATE_TOPIC = Topic.create("Adonis RC Update", AdonisRcUpdateNotifier.class);

    void commands(Map<String, Optional<Command[]>> commands);
}

File: src/main/java/io/stouder/adonis/notifier/AdonisRcUpdateNotifier.kt
package io.stouder.adonis.notifier

import com.intellij.util.messages.Topic
import io.stouder.adonis.cli.json.ace.Command
import java.util.*

interface AdonisRcUpdateNotifier {
    companion object {
        val ADONIS_RC_UPDATE_TOPIC = Topic.create("Adonis RC Update", AdonisRcUpdateNotifier::class.java)
    }

    fun commands(commands: Map<String, Optional<Array<Command>>>?)
}
